src/main/java/bio/terra/service/dataset/DatasetJsonConversion.java

```diff
@@ -52,22 +60,50 @@ public static Dataset datasetRequestToDataset(DatasetRequestModel datasetRequest
                 assetSpecifications.add(assetModelToAssetSpecification(asset, tablesMap, relationshipsMap)));
         }
 
+        CloudPlatform cloudPlatform = Optional.ofNullable(datasetRequest.getCloudPlatform())
+            .orElse(DEFAULT_CLOUD_PLATFORM);
+
+        final List<StorageResource> storageResources;
+        if (cloudPlatform == CloudPlatform.GCP) {
+            storageResources = createGcpStorageResourceValues(datasetRequest.getRegion());
+        } else {
+            throw new UnsupportedOperationException(cloudPlatform + " is not a valid Cloud Platform");
+        }
+
         return new Dataset(new DatasetSummary()
                 .name(datasetRequest.getName())
                 .description(datasetRequest.getDescription())
+                .storage(storageResources)
                 .defaultProfileId(defaultProfileId))
                 .tables(new ArrayList<>(tablesMap.values()))
                 .relationships(new ArrayList<>(relationshipsMap.values()))
                 .assetSpecifications(assetSpecifications);
     }
 
-    public static DatasetSummaryModel datasetSummaryModelFromDatasetSummary(DatasetSummary dataset) {
+    private static List<StorageResource> createGcpStorageResourceValues(String providedRegion) {
+        final GoogleRegion region = Optional.ofNullable(providedRegion)
+            .map(s -> GoogleRegion.fromValue(s.toLowerCase()))
+            .orElse(DEFAULT_GOOGLE_REGION);
+        return Arrays.stream(GoogleCloudResource.values()).map(resource -> {
+            // TODO: Firestore will always be in us-central1 for now, but will likely change in the future.
+            String dbRegion = (resource.equals(GoogleCloudResource.FIRESTORE)) ?
+                DEFAULT_GOOGLE_REGION.toString() :
+                region.toString();
+            return new StorageResource()
+                .cloudPlatform(CloudPlatform.GCP)
+                .region(dbRegion)
+                .cloudResource(resource.toString());
+        }).collect(Collectors.toList());
+    }
+
+    public static DatasetSummaryModel datasetSummaryModelFromDatasetSummary(DatasetSummary datasetSummary) {
```

Comment:
My usual rule of thumb is that static methods which accept class parameters are often better expressed as methods on the object. Could this be moved to `DatasetSummary`, e.g. `DatasetSummary.toModel()`?

Reply:
I agree with you, but I think that's a bit out of scope for this effort. Its pretty weird that we have a one-off class for converting Dataset JSON models to internal models, but we don't have a class for other data types. For example, converting a `SnapshotRequestModel` to `Snapshot` happens in `SnapshotService`. I've created [this ticket](https://broadworkbench.atlassian.net/browse/DR-1790) to capture the work.

Reply:
This is new code that's being added, how is changing it out of scope? There's no need to change all existing methods like this right now, a ticket is a good way to track that work. In the meantime, this code can be done in a better way, if we agree that it's better.

Reply:
I'm trying to keep the amount of code changes constrained to work that supports regions. `DatasetJsonConversion` seems to do a lot and I'm a little daunted by the work of refactoring it out.